caas/kubernetes/provider/k8s.go

```diff
@@ -66,6 +68,8 @@ type kubernetesClient struct {
 	// namespace is the k8s namespace to use when
 	// creating k8s resources.
 	namespace string
+
+	apiextensionsClient apiextensionsclientset.Interface
```

Comment:
Can we move this up next to kubernetes.Interface
Not sure if we want to embed it or not?